### PR TITLE
Enabling pylint's `unused-argument` flag

### LINKED ISSUE #981: Enable pylint's unused-argument flag
### What is happening?

Recently we've been burned a few times because we added an argument to e.g. css.Service.ibmq_compile but forgot to propagate that argument downstream. The unused-argument flag would catch these

unfortunately this would require some modifications, e.g. for unused arguments that are defined in a superclass or inputs automatically generated by the @mock.patch decorator. i believe pylint will let you use underscores as argument names to bypass the check (and either way we can always use the # pylint: disable=unused-argument pragma)

### How can we reproduce the issue?

-

### What should happen?

-

### Environment

- **Superstaq version**: 
- **general-superstaq version**: 
- **cirq-superstaq version**: 
- **qiskit-superstaq version**: 
- **Operating system**: 
- **Python version**: 
- **Any other relevant dependencies**: 

### Any additional context?

_No response_

### PR DESCRIPTION
**Pull request by spencerdearman**
- Updates to include the `unused-argument` flag in the pyproject.toml file
- Updates various files that were flagged when running `./pylint_.py --all`

Closes #981

### CODE REVIEW

**Comment by spencerdearman** [comment]
https://github.com/Infleqtion/server-superstaq/pull/3273

**Comment by spencerdearman** [comment]
#https://github.com/Infleqtion/server-superstaq/pull/3273

File: qiskit-superstaq/qiskit_superstaq/custom_gates.py [hunk 1]
```diff
@@ -370,6 +372,7 @@ def _define(self) -> None:
 
     def __array__(self, dtype: type | None = None) -> npt.NDArray[np.complex_]:
         """Returns a numpy array of the inverse iX gate."""
+        # pylint: disable=unused-argument
         return np.array([[0, -1j], [-1j, 0]])
```
Comment (richrines1) [inline qiskit-superstaq/qiskit_superstaq/custom_gates.py:376]:
```suggestion
        return np.array([[0, -1j], [-1j, 0]], dtype=dtype)
```

File: qiskit-superstaq/qiskit_superstaq/superstaq_provider.py [hunk 2]
```diff
@@ -276,6 +276,7 @@ def aqt_compile_eca(
         Raises:
             ValueError: If `target` is not a valid AQT target.
         """
+        # pylint: disable=unused-argument
```
Comment (richrines1) [inline qiskit-superstaq/qiskit_superstaq/superstaq_provider.py:279]: which argument isn't used here?

File: qiskit-superstaq/qiskit_superstaq/custom_gates.py [hunk 3]
```diff
@@ -67,6 +67,7 @@ def _define(self) -> None:
 
     def __array__(self, dtype: type | None = None) -> npt.NDArray[np.bool_]:
         """Returns an array for the AceCR gate."""
+        # pylint: disable=unused-argument
         return qiskit.quantum_info.Operator(self.definition).data
```
Comment (richrines1) [inline qiskit-superstaq/qiskit_superstaq/custom_gates.py:71]: ideally we should be doing something with the correct precision here, but for now we can use something like
```suggestion
        matrix = qiskit.quantum_info.Operator(self.definition).to_matrix()
        return np.asarray(matrix, dtype=dtype)
```

**Review by vtomole — CHANGES_REQUESTED** [review]
Looks good so far. Now, fix all the `unused arguments` errors and remove the `# pylint: disable=unused-argument`s

**Review by richrines1 — COMMENTED** [review]
(a few notes)

**Comment by spencerdearman** [comment]
I went through and resolved what @richrines1 suggested. Let me know if there is anything I missed/if you have any other changes that I should make.

File: cirq-superstaq/cirq_superstaq/ops/qubit_gates.py [hunk 4]
```diff
@@ -586,7 +586,7 @@ def _json_dict_(self) -> dict[str, Any]:
         return cirq.obj_to_dict_helper(self, ["component_gates"])
 
     @classmethod
-    def _from_json_dict_(cls, component_gates: list[cirq.Gate], **kwargs: Any) -> ParallelGates:
+    def _from_json_dict_(cls, component_gates: list[cirq.Gate], **_kwargs: Any) -> ParallelGates:
```
Comment (richrines1) [inline cirq-superstaq/cirq_superstaq/ops/qubit_gates.py:589]: predates this pr, but while we're here i think this should just be
```suggestion
    def _from_json_dict_(cls, component_gates: list[cirq.Gate], **_kwargs: object) -> ParallelGates:
```

File: cirq-superstaq/cirq_superstaq/ops/qubit_gates.py [hunk 5]
```diff
@@ -826,7 +826,7 @@ def __repr__(self) -> str:
         return f"css.ops.qubit_gates.{str(self)}"
 
     @classmethod
-    def _from_json_dict_(cls, **kwargs: Any) -> IXGate:
+    def _from_json_dict_(cls, **_kwargs: Any) -> IXGate:
         return IXGate()
```
Comment (richrines1) [inline cirq-superstaq/cirq_superstaq/ops/qubit_gates.py:830]:
```suggestion
    def _from_json_dict_(cls, **_kwargs: object) -> IXGate:
        return cls()
```

**Review by richrines1 — COMMENTED** [review]
looks good, just some nits

**Review by richrines1 — APPROVED** [review]
lgtm!

**Review by spencerdearman — COMMENTED** [review]
Fixed issues

**Review by vtomole — APPROVED** [review]
LGTM

**Pull request merged**